test/.eslintrc.yaml

```
@@ -11,6 +11,42 @@ rules:
+    - selector: "CallExpression[callee.property.name='strictEqual'][arguments.2.type='Literal']"
+      message: "Do not use a literal for the third argument of assert.strictEqual()"
+    - selector: "CallExpression:matches([callee.name='throws'], [callee.property.name='throws'])[arguments.1.type='Literal']:not([arguments.1.regex])"
+      message: "Use an object as second argument of `assert.throws()`."
+    - selector: "CallExpression:matches([callee.name='throws'], [callee.property.name='throws'])[arguments.length<2]"
+      message: "`assert.throws()` must be invoked with at least two arguments."
+    - selector: "CallExpression[callee.name='setTimeout'][arguments.length<2]"
+      message: "`setTimeout()` must be invoked with at least two arguments."
+    - selector: "CallExpression[callee.name='setInterval'][arguments.length<2]"
+      message: "`setInterval()` must be invoked with at least two arguments."
+    - selector: 'ThrowStatement > CallExpression[callee.name=/Error$/]'
+      message: "Use `new` keyword when throwing an `Error`."
+    - selector: "CallExpression:matches([callee.name='notDeepStrictEqual'], [callee.property.name='notDeepStrictEqual'])[arguments.0.type='Literal']:not([arguments.1.type='Literal']):not([arguments.1.type='ObjectExpression']):not([arguments.1.type='ArrayExpression']):not([arguments.1.type='UnaryExpression'])"
+      message: "The first argument should be the `actual`, not the `expected` value."
+    - selector: "CallExpression:matches([callee.name='notStrictEqual'], [callee.property.name='notStrictEqual'])[arguments.0.type='Literal']:not([arguments.1.type='Literal']):not([arguments.1.type='ObjectExpression']):not([arguments.1.type='ArrayExpression']):not([arguments.1.type='UnaryExpression'])"
+      message: "The first argument should be the `actual`, not the `expected` value."
+    - selector: "CallExpression:matches([callee.name='deepStrictEqual'], [callee.property.name='deepStrictEqual'])[arguments.0.type='Literal']:not([arguments.1.type='Literal']):not([arguments.1.type='ObjectExpression']):not([arguments.1.type='ArrayExpression']):not([arguments.1.type='UnaryExpression'])"
+      message: "The first argument should be the `actual`, not the `expected` value."
+    - selector: "CallExpression:matches([callee.name='strictEqual'], [callee.property.name='strictEqual'])[arguments.0.type='Literal']:not([arguments.1.type='Literal']):not([arguments.1.type='ObjectExpression']):not([arguments.1.type='ArrayExpression']):not([arguments.1.type='UnaryExpression'])"
+      message: "The first argument should be the `actual`, not the `expected` value."
+    - selector: "CallExpression[callee.name='isNaN']"
+      message: "Use Number.isNaN() instead of the global isNaN() function."
+    - selector: "CallExpression:matches([callee.name='debuglog'], [callee.property.name='debuglog']):not([arguments.0.value='test'])"
+      message: "Only use 'test' as debuglog value inside of the tests folder."
```

Total optional nit-pick but I prefer we standardize on "directory" over "folder":

```suggestion
      message: "Only use 'test' as debuglog value inside of the test directory."
```

---

Or perhaps even simpler:

```suggestion
      message: "Use 'test' as debuglog value in tests."
```